core/modules/node/node.pages.inc

```diff
@@ -883,17 +907,36 @@ function node_revision_overview($node) {
   if ((user_access('delete revisions') || user_access('administer nodes')) && node_access('delete', $node)) {
     $delete_permission = TRUE;
   }
+  $edit_permission = FALSE;
+  if ((user_access('update revisions') || user_access('administer nodes')) && node_access('update', $node)) {
+    $edit_permission = TRUE;
+  }
   foreach ($revisions as $revision) {
     $row = array();
-    if ($revision->current_vid > 0) {
-      $row[] = array('data' => t('!date by !username', array('!date' => l(format_date($revision->timestamp, 'short'), "node/$node->nid"), '!username' => theme('username', array('account' => $revision))))
-                               . (($revision->log != '') ? '<p class="revision-log">' . filter_xss($revision->log) . '</p>' : ''),
-                     'class' => array('revision-current'));
-      $row[] = array('data' => backdrop_placeholder(t('current revision')), 'class' => array('revision-current'));
+    $is_active_revision = ($revision->vid == $node->active_revision);
+    $links = array();
+    $row_class = '';
+    if ($is_active_revision) {
+      // @todo remove the revision-current class in 2.x
+      $row_class = 'revision-active revision-current';
```

Comment:
For an admin-only class like this, it seems we could remove `revision-active` with minimal impact.

Reply:
true